api/src/org/apache/cloudstack/api/command/user/snapshot/CreateSnapshotCmd.java

```diff
@@ -224,8 +224,8 @@ public void execute() {
 
         Snapshot.LocationType locationTypeToReturn = Snapshot.LocationType.values()[0];
 
-        if (locationType != null && locationType >= 0 && locationType < Snapshot.LocationType.values().length) {
-            locationTypeToReturn = Snapshot.LocationType.values()[locationType];
+        if (locationType != null && locationType >= 0 && locationType <= Snapshot.LocationType.values().length) {
```

Comment:
If the user can only specify 1 or 2 (which is currently the case), then locationType >= 0 should be locationType >= 1.

Also, the description for locationType should specify the text 'Standard' instead of 'Primary'.

Currently says: "Currently applicable only for managed storage. " +
            "Valid location types: 1 = 'primary'; 2 = 'archive'. Default = 1."

Should say: "Currently applicable only for managed storage. " +
            "Valid location types: 1 = 'standard'; 2 = 'archive'. Default = 1."